app/services/billable_metrics/prorated_aggregations/base_service.rb

```diff
@@ -105,15 +105,30 @@ def persisted_pro_rata
       end
 
       def per_event_aggregation
+        recurring_value = previous_charge_fee ? previous_charge_fee.units : nil
```

Comment:
Style/SafeNavigation: Use safe navigation (`&.`) instead of checking if an object exists before calling the method.